pkg/oci/client/compute.go

```diff
@@ -106,14 +105,19 @@ func (c *client) GetPrimaryVNICForInstance(ctx context.Context, compartmentID, i
 
 		for _, attachment := range resp.Items {
 			if attachment.LifecycleState != core.VnicAttachmentLifecycleStateAttached {
-				glog.Infof("VNIC attachment %q for instance %q has a state of %q (not %q)",
-					*attachment.Id, instanceID, attachment.LifecycleState, core.VnicAttachmentLifecycleStateAttached)
+				c.logger.With(
+					"instanceID", instanceID,
+					"vnicAttachmentID", *attachment.Id,
+					"expectedLifecycleState", core.VnicAttachmentLifecycleStateAttached,
+					"actualLifecycleState", attachment.LifecycleState,
+				).Info("VNIC attachment in unexpected lifecycle state")
```

Comment:
personally i'd just have the log msg as "VNIC attachment is not in attached state" and remove the expected msg.